EIPS/eip-5341.md

```diff
@@ -0,0 +1,117 @@
+---
+eip: 5341
+title: NFT Referral Program
+description: An interface for defining NFT referral program from which referrers share referral code and earn rebates.
+author: Yuchen Jiang (@cdpiano), Charlie Chen (@chcharcharlie), Yuzhang Wu (@wuyuzhang)
+discussions-to: https://ethereum-magicians.org/t/new-proposal-nft-referral-program/10121
+status: Draft
+type: Standards Track
+category: ERC
+created: 2022-07-25
+requires: 721
+---
+
+## Abstract
+An standard interface for defining NFT referral program from which referrers share referral code and earn a percentage of mint cost from referees who mint by using the referral code.
+
+## Motivation
+In real world we all need to rent or buy a house. The goal of renting or buying a house is not always to trade or invest, but to utilize it as a living space as well. Existing residents get bonus after they invite a new tenant to join, before which they spent time introducing the apartment to potential new tenants and answering their questions from different angles. These existing residents significantly contribute to the success of an apartment rental, building a large and interactive community.
+
+It's likewise for NFT space. Not only team members of a project, it's the influencers and the strong believers of a project who introduce the project to more people and help new people onboard. They are like ambassadors of the project and contribute a lot toe the project, but currently there isn't a good way to incentivize them besides their true passion for the project and community. We would like to propose a standard interface for NFT projects to give referral bonus to these ambassadors.
+
+Enabling more NFT projects to unify on a referral program standard will benefit the entire NFT ecosystem and we can see this makes a larger impact on non-transferrable NFTs. Currently NFT is still considered something money-oriented and some people who mint are just to flip and take profit. However for non-transferrable NFTs the utility attribute will be enhanced and the financial attribute weakened, just like apartment rental where you don't usually flip the apartment but live inside it. Referral program may contribute to the success more in the space of non-transferrable NFTs.
+
+## Specification
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+### Contract Interface
+
+```solidity
+interface INFTReferralProgram {
+    /// @notice Get the referrer address of a tokenId
+    /// @param tokenId The tokend id of the NFT
+    function referrerOf(uint256 tokenId) external view returns (address);
```

Comment:
How is this determined?

Reply:
+1 
Probably this EIP shall also specify a `refer` function to set the information of the referrals to be queried by `referrerOf` and `referralFee`